src/components/form/PillInput.res

```diff
@@ -0,0 +1,179 @@
+      | None => addItem(inputValue)
+      }
+    }
+  }
+
+  let toggleEditingItem = item => {
+    setEditingItem(_ => Some(item))
+    setEditInput(_ => item)
+  }
+
+  <div className="w-full ">
+    <div className="w-full flex flex-wrap gap-2 border p-2 text-sm rounded-md">
+      {items
+      ->Array.mapWithIndex((item, i) =>
+        switch editingItem {
+        | Some(selected) if selected == item =>
+          <div key={Int.toString(i)} className="flex flex-wrap gap-1 p-1 border text-sm rounded-md">
+            <input
+              type_="text"
+              value={editInput}
+              onBlur={_ => saveItem(item)}
+              onInput=handleEditChange
+              onKeyDown={ev => {
+                let key = ev->ReactEvent.Keyboard.key
+                let keyCode = ev->ReactEvent.Keyboard.keyCode
+                if key === "Enter" || keyCode === 13 {
+                  ReactEvent.Keyboard.preventDefault(ev)
+                  saveItem(item)
+                }
+              }}
+              className="rounded-md p-1 text-sm flex-grow"
+            />
+            <button
+              onClick={_ => removeItem(item)} className="ml-2 hover:bg-gray-300 rounded-md p-1">
+              <Icon name="cross-outline" size=14 />
+            </button>
+          </div>
+        | _ =>
+          <div key={Int.toString(i)} className="flex flex-wrap gap-1 p-1 border text-sm rounded-md">
+            <div
+              className="cursor-pointer rounded-md px-2 py-1 text-sm text-gray-600 flex-grow"
+              onClick={_ => toggleEditingItem(item)}>
+              {React.string(item)}
+            </div>
+            <button
+              onClick={_ => removeItem(item)} className="ml-2 hover:bg-gray-300 rounded-md p-1">
+              <Icon name="cross-outline" size=14 />
+            </button>
+          </div>
+        }
+      )
+      ->React.array}
+      <div className="relative">
+        <input
+          type_="text"
+          value={inputValue}
+          placeholder
+          onChange=handleInputChange
+          onKeyDown=handleKeyDown
+          className="max-w-fit outline-none p-2 flex-grow"
+          name
+        />
+        <RenderIf condition={suggestion->Option.isSome}>
+          <div
+            onClick={_ => handleSuggestionClick()}
+            className="absolute z-10  w-full min-w-80 bg-white border border-gray-300 rounded-md shadow-lg mt-1 cursor-pointer top-10 h-16">
+            <div className="bg-gray-200 w-full h-[calc(100%-16px)] my-2 flex items-center px-4">
+              <div className="flex items-center gap-2">
+                <img alt="user_icon" src={`/icons/user_icon.svg`} className="h-6 w-6" />
+                <span className="font-medium"> {React.string(suggestion->Option.getOr(""))} </span>
+              </div>
+            </div>
+          </div>
+        </RenderIf>
+      </div>
+    </div>
+    <RenderIf condition={!(error->LogicUtils.isEmptyString)}>
```

Comment:
`isNonEmptyString` can be used directly, and logicUtils can be opend at once at once if its used inmultiple places

Reply:
It's used in just 2 places.